### ISSUE TITLE
feat: training and LoRA support for 4B DiT models

### ISSUE THREAD

**Issue by ChuxiJ** [ace-step/ACE-Step-1.5#994]
## Summary

Enable LoRA training on 4B DiT models. The training infrastructure currently assumes 2B model dimensions for VRAM estimation and has hardcoded weight offsets. These need updating for 4B's larger parameter count.

## Background

4B DiT architecture:
- Decoder: hidden_size=2560, 32 layers, 32 attention heads → ~2x parameters vs 2B
- Encoder: hidden_size=2048, 8 layers (same as 2B)
- LoRA applies to decoder attention projections (q_proj, k_proj, v_proj, o_proj)
- More layers (32 vs 24) means more LoRA-injectable modules

### 1. `acestep/training_v2/gpu_utils.py` — Per-batch VRAM Estimate

**Line 157-160**: Hardcoded for 2B (24 layers, hidden_size=2048):
```python
# Rough per-batch memory estimate for a single forward+backward pass through
# the full DiT decoder (24 layers, hidden_size=2048) with all parameters
# requiring gradients.  This is a *very* conservative upper bound.
_BYTES_PER_BATCH_ESTIMATE_BF16: float = 1200.0  # ~1.2 GiB per sample
```

**Line 216-217**: Fixed 4GB model weights offset:
```python
usable_mb = max(0.0, usable_mb - 4096.0)  # Subtract ~4 GiB for the model weights themselves
```

**Required changes:**
- Make `_BYTES_PER_BATCH_ESTIMATE_BF16` model-size-aware (~1.8-2.4 GiB for 4B)
- Make weight offset model-size-aware (~8 GiB for 4B in bf16)
- Options:
  - a) Read `hidden_size` and `num_hidden_layers` from model config to compute estimate
  - b) Add a `_BYTES_PER_BATCH_ESTIMATE_XL_BF16` constant
  - c) Pass model_size parameter to `estimate_batch_budget()`

### 2. `acestep/training_v2/model_loader.py` — Variant Directory

**Lines 54-58**: Currently only 2B variant directories:
```python
_VARIANT_DIR = {
    "turbo": "acestep-v15-turbo",
    "base": "acestep-v15-base",
    "sft": "acestep-v15-sft",
}
```

**Required**: Add 4B entries if using separate directory names:
```python
_VARIANT_DIR = {
    "turbo": "acestep-v15-turbo",
    "base": "acestep-v15-base",
    "sft": "acestep-v15-sft",
    "xl-base": "acestep-v15-xl-base",
    "xl-sft": "acestep-v15-xl-sft",
    "xl-turbo": "acestep-v15-xl-turbo",
}
```

Or extend the resolution logic to accept 4B model paths directly.

### 3. `acestep/training_v2/model_discovery.py` — Base Defaults

**Lines 24-28**: May need 4B-specific timestep/shift defaults if they differ:
```python
_BASE_DEFAULTS = {
    "turbo": {"is_turbo": True, "timestep_mu": -0.4, ...},
    "base":  {"is_turbo": False, "timestep_mu": -0.4, ...},
    "sft":   {"is_turbo": False, "timestep_mu": -0.4, ...},
}
```

These values are also read from the model's `config.json` at runtime, so this may be fine if 4B checkpoints include correct values.

### 4. `acestep/training/lora_injection.py` — Target Module Detection

**Lines 116-133**: Dynamically discovers LoRA targets:
```python
def get_dit_target_modules(model) -> List[str]:
    for name, module in model.decoder.named_modules():
        if any(proj in name for proj in ["q_proj", "k_proj", "v_proj", "o_proj"]):
            if isinstance(module, nn.Linear):
                target_modules.append(name)
```

This should work for 4B **if** 4B uses the same attention projection names. 

- [ ] **Verify**: Load 4B model and print `get_dit_target_modules(model)` to confirm correct discovery

### 5. `acestep/training_v2/presets/*.json` — Training Presets

Current presets assume 2B DiT. For 4B, effective LoRA capacity at the same rank is halved (because the model has 2x parameters). Recommended adjustments:

| Preset | Current Rank | 4B Suggested Rank | Rationale |
|--------|-------------|-------------------|-----------|
| vram_8gb | 16 | N/A (can't train 4B) | Insufficient VRAM |
| vram_12gb | 32 | N/A or 16 (marginal) | Very tight VRAM |
| vram_16gb | 64 | 32-48 | Need room for 4B weights |
| vram_24gb_plus | 128 | 64-96 | Scale rank proportionally |
| recommended | 64 | 48-64 | Balance quality/VRAM |
| high_quality | 128 | 96-128 | Maximum quality |
| quick_test | 16 | 16 | Quick validation |

Options:
- a) Create separate `*_xl.json` preset files
- b) Make presets model-size-aware (dynamic rank scaling)
- c) Add a size multiplier in the training config

### 6. `acestep/training_v2/estimate.py` — Sensitivity Analysis

**Lines 76-81**: Reads timestep params from model config — already model-aware ✓

**Lines 236-261**: `_find_attention_modules()` uses dynamic discovery — already works ✓

But batch budget estimation needs the same fixes as gpu_utils.py (#1 above).

## Acceptance Criteria

- [ ] `get_dit_target_modules()` correctly discovers all attention modules in 4B model
- [ ] Training runs without OOM using auto-configured batch size
- [ ] VRAM estimation in training wizard is accurate for 4B
- [ ] LoRA training produces quality improvements (validation loss decreases)
- [ ] LoRA checkpoints from 4B training can be loaded for inference
- [ ] Existing 2B training workflow is unaffected
